test/test-core.js

```diff
@@ -247,6 +247,180 @@
 		});
 	});
 
+	t('Dynamic response status callback - list of statuses as an array (errors only)', function(assert) {
+		var done = assert.async();
+		var possibleStatuses = [500,400,404];
+	  	var returnedStatuses = [];
+		var maxNumLoops = possibleStatuses.length * 50;
+		var numLoops = 0;
+		var numLoopsComplete = 0;
+
+		$.mockjax({
+			url: '/response-callback',
+			response: function() {
+				this.status = possibleStatuses;
+				this.statusText = 'Internal Server Error';
+			}
+		});
+		
+		var completeCallback = function(xhr) {
+			assert.notEqual($.inArray(xhr.status, possibleStatuses), -1, 'Dynamically set random response status found');
+
+			if( $.fn.jquery !== '1.5.2') {
+				// This assertion fails in 1.5.2 due to this bug: http://bugs.jquery.com/ticket/9854
+				// The statusText is being modified internally by jQuery in 1.5.2
+				assert.equal(xhr.statusText, 'Internal Server Error', 'Dynamically set response statusText matches');
```

Comment:
This is only the case if the status code is 500, I think. Since we're not evaluating the status text otherwise I'd just kill this whole conditional.